assessWorkflows.java

```diff
@@ -77,35 +129,44 @@ private void analyzeWorkflow(GHContent workflow) {
         }
         try {
             System.out.println(" 👀 "+workflow.getHtmlUrl());
-            var content = IOUtils.toString(workflow.read(), "UTF-8");
-            content.lines().filter(l -> !l.startsWith("#") && l.contains("uses:"))
-                            .map(this::toAction)
-                            .filter(Objects::nonNull)
-                            .filter(this::isNotTrusted)
-                            .forEach(a -> System.out.println("\t"+a));
-            //System.out.println(content);
+            var mapper = new ObjectMapper(new YAMLFactory()).configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            Workflow wf = null;
+            try {
+                wf = mapper.readValue(IOUtils.toString(workflow.read(), "UTF-8"), Workflow.class);
+            } catch (MismatchedInputException ex) {
+                // non parsable yaml / skipping
+                wf = new Workflow("", Map.of());
+            }
+
+            wf.jobs().forEach((k, v) -> {
+                if (v.steps() != null) {
+                    v.steps().forEach(s -> {
+                        if (!s.isTrusted()) {
+                            try {
+                                var repo = github.getRepository(s.getOrg() + "/" + s.getRepo());
+                                var sha = "";
+                                try {
+                                    var tree = repo.getTree(s.getVersion());
+                                    sha = tree.getSha();
+                                } catch (Exception e) {
+                                    System.out.println("Job " + k + " is using action " + s.uses() + " but we cannot identify the commit SHA");
+                                }
```

Comment:
this should be cached, since several repositories point at the same action, no need to recompute the SHA1 several times

Reply:
ok, done.